api/editions.go

```diff
@@ -16,6 +17,22 @@ func (api *DatasetAPI) getEditions(w http.ResponseWriter, r *http.Request) {
 	datasetID := vars["dataset_id"]
 	logData := log.Data{"dataset_id": datasetID}
 
+	// get limit from query parameters, or default value
+	limit, err := utils.GetPositiveIntQueryParameter(r.URL.Query(), "limit", api.defaultLimit)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
```

Comment:
It is best practice to return log.Error(err) and log.Data or in this case logData in the log event as previously stated